### PR TITLE
to resolve Issue 16

### LINKED ISSUE #16: Use Pydantic V2 models
**Is your feature request related to a problem? Please describe**

Use Pydantic V2 `BaseModel` class rather than V1 (maintenance purpose only).

**Describe the solution you'd like**

Migrate referring the Pydantic V2 [Migration Guide](https://docs.pydantic.dev/latest/migration/).

**Describe alternatives you've considered**

Use Pydantic V2/V1 backward compatibility:

```python
try:
    from pydantic import v1 as pydantic
except ImportError:
    import pydantic  # type: ignore
```

### PR DESCRIPTION
**Pull request by roman2git**
### What?
To close issue #16

### Have you done?
- [X] Code tests
- [X] Update documentation
- [ ] Update [changelog](https://github.com/MAIF/arta/blob/main/CHANGELOG.md)

### Details to be checked: (optional)

There are minor changes which were highlighted by linter/IDE:
- Added class fields to rule/Rule
- fixed field in docstring 

### Linked issues: (optional)
- Close #16

### CODE REVIEW

**Review by develop-cs — COMMENTED** [review]
Thank you for your contribution 👍 
Few comments following the [Conventional Comments](https://conventionalcomments.org/) format.

File: src/arta/models.py [hunk 1]
```diff
@@ -3,74 +3,153 @@
...
+        """Pydantic model for validating configuration files."""
 
-class Configuration(pydantic.BaseModel):
-    """Pydantic model for validating configuration files."""
+        conditions: Optional[Dict[str, Condition]] = None
+        conditions_source_modules: Optional[List[str]] = None
+        actions_source_modules: List[str]
+        custom_classes_source_modules: Optional[List[str]] = None
+        condition_factory_mapping: Optional[Dict[str, str]] = None
+        rules: Dict[str, Dict[str, Dict[str, RulesConfig]]]
```
[the remark this answers is not in the document]
Comment (roman2git) [reply src/arta/models.py:63]: pydantic.constr  is discouraged in v2 and will be deprecated in v3
https://docs.pydantic.dev/2.9/api/types/#pydantic.types.constr

Its successors is Annotated with StringConstraints (see line 55 in the models)
However such definition does not work with datatype like (there were an error - don't remember exactly as well as an issue on pydantic github - don't have it handy) 
That is why I added usual validation func for the same - ensure uppercase string for the key

[the remark this answers is not in the document]
Comment (develop-cs) [reply src/arta/models.py:63]: Good to know, thank you for the details.

**Review by develop-cs — APPROVED** [review]
Great 👍 Thank you

**Pull request merged**